src/Provider/Activation.php

```diff
@@ -37,6 +38,33 @@ public function register_hooks() {
 	 * @since 0.3.0
 	 */
 	public function activate() {
+		$this->setup_storage();
 		update_option( 'satispress_flush_rewrite_rules', 'yes' );
 	}
+
+	/**
+	 * Set up the local storage provider.
+	 *
+	 * Creates the cache path and adds an .htaccess file to prevent HTTP access
+	 * on Apache.
+	 *
+	 * @since 0.3.0
+	 */
+	public function setup_storage() {
+		$container  = $this->plugin->get_container();
+		$cache_path = $container->get( 'cache.path' );
+		$storage    = $container->get( 'storage' );
+		$htaccess   = $container->get( 'htaccess.handler' );
+
+		if ( ! $storage instanceof Local ) {
```

Comment:
This feels like a code smell. If the storage was AWS S3, would it not have its own setup?